packages/client/src/pages/measurables/MeasurableIndex__Logic.re

```diff
@@ -6,7 +6,30 @@ module ReducerConfig = {
     states: option(array(Types.measurableState)),
   };
 
-  let getId = (params: itemType) => params.id;
+  let getId = (itemType: itemType) => itemType.id;
+
+  let onItemDeselected = (params: callFnParams) => {
+    let st =
+      params.states
+      |> MeasurableQueryIndex.make
+      |> MeasurableQueryIndex.toUrlParams;
+    let channelLink = Routing.Url.toString(ChannelShow(params.channelId));
+    ReasonReact.Router.replace(channelLink ++ st ++ "#!");
```

Comment:
What's the point of the "#!"?

Reply:
To make it possible to differentiate it between two types of events. Which are when you push something to browser's history and another one when you replace URL.